plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -84,6 +250,57 @@ private function is_latest_woocommerce_version() {
 		return version_compare( WC()->version, $latest_wc_version, '>=' );
 	}
 
+	/**
+	 * Check if the error exclusively contains third-party stack frames for fatal-errors source context.
+	 *
+	 * @param string $message The error message.
+	 * @param array  $context The error context.
+	 *
+	 * @return bool
+	 */
+	protected function is_third_party_error( string $message, array $context ): bool {
+		// Only check for fatal-errors source context.
+		if ( ! isset( $context['source'] ) || 'fatal-errors' !== $context['source'] ) {
+			return false;
+		}
+
+		// If backtrace is not available, we can't determine if the error is third-party. Log it for further investigation.
+		if ( ! isset( $context['backtrace'] ) || ! is_array( $context['backtrace'] ) ) {
+			return false;
+		}
+
+		$wc_plugin_dir = StringUtil::normalize_local_path_slashes( WC_ABSPATH );
+
+		// Check if the error message contains the WooCommerce plugin directory.
+		if ( is_string( $message ) && str_contains( $message, $wc_plugin_dir ) ) {
+			return false;
+		}
+
+		// Check if the backtrace contains the WooCommerce plugin directory.
+		foreach ( $context['backtrace'] as $trace ) {
+			if ( is_string( $trace ) && str_contains( $trace, $wc_plugin_dir ) ) {
+				return false;
+			}
+
+			if ( is_array( $trace ) && isset( $trace['file'] ) && str_contains( $trace['file'], $wc_plugin_dir ) ) {
+				return false;
+			}
+		}
+
+		$is_third_party_error = true;
+
+		/**
+		 * Filter to allow other plugins to overwrite the result of the third-party error check for remote logging.
+		 *
+		 * @since 9.2.0
+		 *
+		 * @param bool   $is_third_party_error The result of the third-party error check.
+		 * @param string $message              The error message.
+		 * @param array  $context              The error context.
+		 */
+		return apply_filters( 'woocommerce_remote_logging_is_third_party_error', $is_third_party_error, $message, $context );
```

Comment:
Again, completely optional change you can feel free to skip:

```suggestion
		/**
		 * Filter to allow other plugins to overwrite the result of the third-party error check for remote logging.
		 *
		 * @since 9.2.0
		 *
		 * @param bool   $is_third_party_error The result of the third-party error check.
		 * @param string $message              The error message.
		 * @param array  $context              The error context.
		 */
		return apply_filters( 'woocommerce_remote_logging_is_third_party_error', true, $message, $context );
```

We don't need to assign the value-to-be-filtered to a variable, since it is basically a constant (but if you prefer to do it this way for readability, totally fine).